Imagine/Filter/PostProcessor/CwebpPostProcessor.php

```diff
@@ -0,0 +1,230 @@
+     * one which gives the smaller size. The **fast** mode will just try to form an a priori guess without testing all
+     * modes.
+     *
+     * @var string
+     */
+    protected $alphaFilter;
+
+    /**
+     * Specify the algorithm used for alpha compression: **0** or **1**. Algorithm **0** denotes no compression, **1**
+     * uses WebP lossless format for compression.
+     *
+     * @var int
+     */
+    protected $alphaMethod;
+
+    /**
+     * Preserve RGB values in transparent area. The default is off, to help compressibility.
+     *
+     * @var bool
+     */
+    protected $exact;
+
+    /**
+     * A comma separated list of metadata to copy from the input to the output if present. Valid values: **all**,
+     * **none**, **exif**, **icc**, **xmp**.
+     *
+     * Note that each input format may not support all combinations.
+     *
+     * @var string
+     */
+    protected $metadata;
+
+    public function __construct(
+        string $executablePath = '/usr/bin/cwebp',
+        string $temporaryRootPath = null,
+        int $nearLossless = null,
+        int $q = null,
+        int $alphaQ = null,
+        int $m = null,
+        string $alphaFilter = null,
+        int $alphaMethod = null,
+        bool $exact = null,
+        string $metadata = null
+    ) {
+        parent::__construct($executablePath, $temporaryRootPath);
+
+        $this->nearLossless = $nearLossless;
+        $this->q = $q;
+        $this->alphaQ = $alphaQ;
+        $this->m = $m;
+        $this->alphaFilter = $alphaFilter;
+        $this->alphaMethod = $alphaMethod;
+        $this->exact = $exact;
+        $this->metadata = $metadata;
+    }
+
+    public function process(BinaryInterface $binary, array $options = []): BinaryInterface
+    {
+        if (!$this->isBinaryTypeWebpImage($binary)) {
+            return $binary;
+        }
+
+        $file = $this->writeTemporaryFile($binary, $options, 'imagine-post-processor-cwebp');
+        $arguments = $this->getProcessArguments($options);
+        $arguments[] = $file;
+        $arguments[] = '-o';
+        $arguments[] = '-';
+        $process = $this->createProcess($arguments, $options);
+
+        $process->run();
+
+        if (!$this->isSuccessfulProcess($process)) {
+            unlink($file);
+
+            throw new ProcessFailedException($process);
+        }
+
+        $result = new Binary($process->getOutput(), $binary->getMimeType(), $binary->getFormat());
+
+        unlink($file);
+
+        return $result;
+    }
+
+    protected function isBinaryTypeWebpImage(BinaryInterface $binary): bool
+    {
+        return $this->isBinaryTypeMatch($binary, ['image/webp']);
+    }
+
+    /**
+     * @param int|string[] $options
+     *
+     * @return string[]
+     */
+    private function getProcessArguments(array $options = []): array
+    {
+        $arguments = [$this->executablePath];
+
```

Comment:
this method could use the symfony OptionsResolver class to be quite a bit more consise, i think. other processors had fewer options, so it showed less. do you want to try that?

Reply:
Yep, will definitely try it. It was my first idea, but I change my mind to be consistent with others processors.

Reply:
Here we go, done!